### PR TITLE
Guard remote branch exists in pull-request:create

### PR DESCRIPTION
**Pull request by sstok**
|Q            |A   |
|---          |--- |
|Fixed Tickets|#407|
|License      |MIT |
                    

Guard the remote branch exists in `pull-request:create`

* check if the remote branch exists
* ask to push when the remote branch doesn't exists but does local
* error when the remote branch doesn't exist and not local or not pushed

TODO:

* [x] Update test
* [x] Fix with partial-matching branch name "407-pull-" matches in "407-pull-create-warn-when-remote-branch-doesn-t-exist-and-ask-to-push" but does not exist itself.

### CODE REVIEW

File: src/Helper/GitHelper.php [hunk 1]
```diff
@@ -199,6 +199,31 @@ function ($commitHash) {
...
+     * @param string $remote Remote name or git-repository url
+     * @param string $branch
+     *
+     * @return bool
+     */
+    public function remoteBranchExists($remote, $branch)
+    {
+        $result = $this->processHelper->runCommand(['git', 'ls-remote', $remote], true);
+
+        return 1 === preg_match('#(?<=\s)'.preg_quote('refs/heads/'.$branch, '#').'(?!\w)#', $result);
```
Comment (cordoval) [inline src/Helper/GitHelper.php]: interesting foo :full_moon_with_face:

File: src/Command/PullRequest/PullRequestCreateCommand.php [hunk 2]
```diff
@@ -123,9 +132,11 @@ protected function execute(InputInterface $input, OutputInterface $output)
     {
         $org = $input->getOption('org');
         $issueNumber = $input->getOption('issue');
+        $template = $input->getOption('template');
+
         $sourceOrg = $input->getOption('source-org');
+        $sourceRepo = $input->getOption('source-repo') ?: $input->getOption('repo') ;
```
Comment (cordoval) [inline src/Command/PullRequest/PullRequestCreateCommand.php]: whitespace escapes you but it is ok, i will merge and fix in a later commit

File: src/Command/PullRequest/PullRequestCreateCommand.php [hunk 3]
```diff
@@ -221,4 +233,57 @@ protected function execute(InputInterface $input, OutputInterface $output)
...
+
+            // This ensures the connection is possible, but does not ensure access is granted.
+            $gitHelper->remoteUpdate($org);
+
+            if (!$input->isInteractive()) {
+                $gitHelper->pushToRemote($org, $branch, true);
+
+                $styleHelper->success(sprintf('Branch "%s" was pushed to "%s".', $branch, $org));
+
+                return ;
```
Comment (cordoval) [inline src/Command/PullRequest/PullRequestCreateCommand.php]: same here

**Comment by cordoval** [comment]
I was not sure about this one because it bloats the command and the method private probably should go to the githelper? maybe not, but just saying. In any case because this is used a lot i am good for usability so merging...

**Comment by cordoval** [comment]
great job @sstok

File: src/Helper/GitHelper.php (hunk 1 above)
Comment (sstok) [inline src/Helper/GitHelper.php]: I needed the look(behind,ahead) because else it would match also match a part of the branch (see description above).

**Pull request merged**